securedrop/journalist_app/api.py

```diff
@@ -279,6 +280,26 @@ def get_current_user():
         user = get_user_object(request)
         return jsonify(user.to_json()), 200
 
+    @api.route('/account/new-password', methods=['POST'])
+    @token_required
+    def new_password():
+        user = get_user_object(request)
+        new_password = json.loads(request.data)['new_password']
+
+        try:
+            user.set_password(new_password)
+        except (InvalidPasswordLength, NonDicewarePassword) as e:
+            return jsonify({'message': str(e)}), 400
+        except Exception as e:
+            return jsonify({'message': 'An error occurred while setting the password. Password unchanged'}), 500
+
+        try:
+            db.session.commit()
+        except Exception as e:
+            return jsonify({'message': 'An error occurred on database commit. Password unchanged.'}), 500
```

Comment:
I would also suggest that all 500 errors on this endpoint use the same string to not give additional information to an attacker.

Reply:
Done